ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/kubernetes_cluster_deploy.py

```diff
@@ -2,144 +2,182 @@
 from Jumpscale.servers.gedis.GedisChatBot import StopChatFlow
 
 
-def chat(bot):
-    """
-    """
-    user_form_data = {}
-    user_info = bot.user_info()
-    user_form_data["chatflow"] = "kubernetes"
-    model = j.threebot.packages.tfgrid_solutions.tfgrid_solutions.bcdb_model_get("tfgrid.solutions.kubernetes.1")
-
-    j.sal.reservation_chatflow.validate_user(user_info)
-
-    bot.md_show("# This wizard will help you deploy a kubernetes cluster")
-    network = j.sal.reservation_chatflow.network_select(bot, j.me.tid)
-    if not network:
-        return
-    currency = network.currency
-    user_form_data["Solution name"] = j.sal.reservation_chatflow.solution_name_add(bot, model)
-    retry = False
-    while True:
-        form = bot.new_form()
-        sizes = ["1 vCPU 2 GiB ram 50GiB disk space", "2 vCPUs 4 GiB ram 100GiB disk space"]
-        cluster_size_string = form.drop_down_choice(
-            "Choose the size of your nodes", sizes, default=sizes[0], retry=retry
+class kubernetesDeploy(j.servers.chatflow.get_class()):
+    steps = [
+        "deployment_start",
+        "network_selection",
+        "solution_name",
+        "nodes_selection",
+        "public_key_get",
+        "expiration_datetime",
+        "ip_selection",
+        "overview",
+        "cluster_reservation",
+        "success",
+    ]
+
+    @j.baseclasses.chatflow_step(title="")
+    def deployment_start(self):
+        user_info = self.user_info()
+        j.sal.reservation_chatflow.validate_user(user_info)
+
+        self.user_form_data = {}
+        self.user_form_data["chatflow"] = "kubernetes"
+        self.md_show("# This wizard will help you deploy a kubernetes cluster")
+
+    @j.baseclasses.chatflow_step(title="Network")
+    def network_selection(self):
+        self.network = j.sal.reservation_chatflow.network_select(self, j.me.tid)
+        if not self.network:
+            return
```

Comment:
return here will not be enough anymore. I guess we need to call self.stop("")

Reply:
since we have bot.stop() in the network_select(..) when no networks are found already, we can remove this check

Reply:
Oh the network_select already does a stop this codepath will not be reached can those 2 lines can indeed just be deleted

Reply:
https://github.com/threefoldtech/jumpscaleX_threebot/commit/7acbef7d8030b79316639179284a39c272825abe#diff-a5383166a7a1c1b6bd81d1638e7492c9L31